### PR TITLE
[Codegen] Handle Duplicative Nodes Class Names

### PR DESCRIPTION
**Pull request by noanflaherty**
We need to ensure that we generate unique node class names during codegen.

This PR does that by upgrading the version of the fern python ast module where fixes have been implemented to support just this.

### CODE REVIEW

File: ee/codegen/src/__test__/__snapshots__/workflow.test.ts.snap [hunk 1]
```diff
@@ -201,3 +201,14 @@ class TestWorkflow(BaseWorkflow[Inputs, BaseState]):
...
 `;
+
+exports[`Workflow > write > should handle the case of multiple nodes with the same label 1`] = `
+"from vellum.workflows import BaseWorkflow
+from .nodes.templating_node import TemplatingNode
+from .nodes.templating_node_1 import TemplatingNode as TemplatingNode1TemplatingNode
+
+
+class TestWorkflow(BaseWorkflow):
+    graph = TemplatingNode >> TemplatingNode
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/__snapshots__/workflow.test.ts.snap]: We aren't yet using the correct aliased class names here. This requires a fern-side fix, which is up for review here: https://github.com/fern-api/fern/pull/5401

File: ee/codegen/src/__test__/__snapshots__/workflow.test.ts.snap [hunk 2]
```diff
@@ -201,3 +201,14 @@ class TestWorkflow(BaseWorkflow[Inputs, BaseState]):
     graph = SearchNode
 "
 `;
+
+exports[`Workflow > write > should handle the case of multiple nodes with the same label 1`] = `
+"from vellum.workflows import BaseWorkflow
+from .nodes.templating_node import TemplatingNode
+from .nodes.templating_node_1 import TemplatingNode as TemplatingNode1TemplatingNode
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/__snapshots__/workflow.test.ts.snap:208]: Here you can see we use unique classname aliases in the import paths.

File: ee/codegen/src/__test__/__snapshots__/workflow.test.ts.snap [hunk 3]
```diff
@@ -201,3 +201,14 @@ class TestWorkflow(BaseWorkflow[Inputs, BaseState]):
...
+
+
+class TestWorkflow(BaseWorkflow):
+    graph = TemplatingNode >> TemplatingNode1TemplatingNode
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/__snapshots__/workflow.test.ts.snap:212]: And here the alias is used in the graph itself

File: ee/codegen/README.md [hunk 4]
```diff
@@ -118,12 +118,12 @@ changes in the workflow codegen package locally, follow these steps:
...
    It should be the relative file path to the `ast` directory in this repository.
    ```json
-   "@fern-api/python-ast": "file:../../../fern-api/fern/generators/python-v2/ast/lib",
+   "@fern-api/python-ast": "file:../../../../fern-api/fern/generators/python-v2/ast/lib",
    ```
 3. Run `npm install` to install the updated dependency.
 4. Navigate to the `fern` repo and be sure to pull the `vargas/publish-python-ast` branch.
 5. Make changes to the fern python ast package.
-6. After making changes, run `pnpm dist` within `fern/generators/python-v2/ast` to compile the changes.
+6. After making changes, run `pnpm compile` within `fern/generators/python-v2/ast` to compile the changes.
```
Comment (dvargasfuertes) [inline ee/codegen/README.md:126]: [q] I'm pretty sure we want `dist` - `dist` more closely resembles what we eventually publish. Did you find an inconsistency?

**Review by dvargas92495 — APPROVED** [review]

File: ee/codegen/README.md (hunk 4 above)
Comment (noanflaherty) [reply ee/codegen/README.md:126]: That command doesn't exist in the `package.json` file of that repo.

Comment (dvargasfuertes) [reply ee/codegen/README.md:126]: Oh whoops, was only on the publishing branch: https://github.com/fern-api/fern/pull/5100/files#diff-0f76e83bc28496aa6a2113d25bf783495f41e86187954ed59aafd574195035ccR28

**Pull request merged**